scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -1519,10 +1517,15 @@
                      ;; TODO: Rescind the offer in fenzo
                      )
     (framework-message [this driver executor-id slave-id message]
-                       ;; the executor-id is the same as the task-id (see cook.mesos.task/task-info->mesos-message)
-                       (async-in-order-processing (:value executor-id)
-                                                  #(handle-framework-message conn handle-progress-message message))
-                       (heartbeat/notify-heartbeat heartbeat-ch executor-id slave-id message))
+                       (try
+                         (let [{:strs [task-id] :as parsed-message} (json/read-str (String. ^bytes message "UTF-8"))]
+                           (async-in-order-processing
+                             task-id #(handle-framework-message conn handle-progress-message parsed-message))
+                           (when (contains? parsed-message "timestamp")
```

Comment:
Should we add some comments talking about the where we expect these messages to come from, why we have the condition, etc?